src/lib.rs

```diff
@@ -3,86 +3,114 @@ extern crate dotenv_codegen;
 
-static MONGODB: OnceCell<Database> = OnceCell::new();
+static MONGODB: OnceCell<RwLock<Option<Database>>> = OnceCell::new();
 
 #[arma]
 fn init() -> Extension {
-    let mut builder = Builder::from_default_env();
+    let mut builder = env_logger::Builder::from_default_env();
     builder.target(Target::Stdout);
     builder.filter_level(LevelFilter::Info);
     builder.init();
-
+    // We unwrap as the cell should not be initialized
+    MONGODB.set(RwLock::new(None)).unwrap();
     Extension::build().command("log", log).finish()
 }
 
-async fn connect() {
-    if MONGODB.get().is_some() {
-        log::warn!(target: "fp_extension", "Connection to DB already present!");
-        return;
-    }
-
-    log::info!(target: "fp_extension", "Connecting to DB!");
+fn connect() -> Result<(), ()> {
+    // We unwrap as the cell should be initialized by now.
+    let lock = MONGODB.get().unwrap();
 
-    let _url = env::var("FP_EXTENSION_MONGO_DB_URL")
-        .unwrap_or_else(|_| dotenv!("FP_EXTENSION_MONGO_DB_URL").to_string());
-    let _db_name = env::var("FP_EXTENSION_MONGO_DB_DBNAME")
-        .unwrap_or_else(|_| dotenv!("FP_EXTENSION_MONGO_DB_DBNAME").to_string());
+    {
+        // In this scope we hold shared access to the lock.
+        let read_l = lock.read().unwrap();
+        if read_l.as_ref().is_some() {
+            log::warn!(target: "fp_extension", "Connection to DB already present!");
+            return Ok(());
+        }
+    }
 
-    if let Ok(client_options) = ClientOptions::parse(_url).await {
-        // client_options.app_name = Some("FPArma Server Extension".to_string());
-        if let Ok(client) = Client::with_options(client_options) {
-            let _ = MONGODB.set(client.database(&_db_name[..]));
-            log::info!(target: "fp_extension", "Connected to DB!");
+    {
+        // In this scope we hold exclusive access to the lock.
+        let mut write_l = lock.write().unwrap();
+        if write_l.as_mut().is_some() {
+            Ok(())
+        } else {
+            log::info!(target: "fp_extension", "Connecting to DB!");
+
+            let url = env::var("FP_EXTENSION_MONGO_DB_URL")
+                .unwrap_or_else(|_| dotenv!("FP_EXTENSION_MONGO_DB_URL").to_string());
+            let db_name = env::var("FP_EXTENSION_MONGO_DB_DBNAME")
+                .unwrap_or_else(|_| dotenv!("FP_EXTENSION_MONGO_DB_DBNAME").to_string());
+
+            match ClientOptions::parse(url) {
+                Ok(client_options) => match Client::with_options(client_options) {
+                    Ok(client) => {
+                        let db = client.database(&db_name[..]);
+                        write_l.replace(db);
+                        log::info!(target: "fp_extension", "Connected to DB!");
+                        Ok(())
+                    }
+                    Err(err) => {
+                        log::error!(target: "fp_extension", "Error connecting to DB: {}", err);
```

Comment:
in general:
error = program about to full crash
warn = program not working correctly but no crash

can you change this to warn?